### ISSUE TITLE
`HashSet`

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#8498]
Continue the draft in #8055
also inspiration https://github.com/mlutze/flixball/tree/hashcollections/src/Utils
also discussion #4519

**Comment by magnus-madsen** [comment]
Fixed in master.

**Comment by JonathanStarup** [comment]
There is no `HashSet`, only `MutHashSet`

**Comment by magnus-madsen** [comment]
You want an immutable one? But does that even exist?

**Comment by magnus-madsen** [comment]
Can you update the title and description, otherwise I will probably just close it again at some point in the future :P

**Comment by JonathanStarup** [comment]
> You want an immutable one? But does that even exist?

Yes, both with fancy and simple implementation https://www.scala-lang.org/api/2.12.3/scala/collection/immutable/HashMap.html

**Comment by magnus-madsen** [comment]
I think this should be left to a library. As far as I understand there is not really a thing such as an immutable hash map.
